guide/students/student_management_service.bal

```diff
@@ -0,0 +1,322 @@
+// Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/io;
+import ballerina/log;
+import ballerina/mysql;
+import ballerina/observe;
+import ballerina/runtime;
+
+// Type Student is created to store details of a student.
+type Student record {
+    int id;
+    int age;
+    string name;
+    int mobNo;
+    string address;
+};
+
+// End point for marks details client.
+endpoint http:Client marksService {
+    url: " http://localhost:9191"
+};
+
+// Endpoint for mysql client.
+endpoint mysql:Client testDB {
+    host: "localhost",
+    port: 3306,
+    name: "testdb",
+    username: "root",
+    password: "",
+    poolOptions: { maximumPoolSize: 5 },
+    dbOptions: { useSSL: false }
+};
+
+// This service listener.
+endpoint http:Listener listener1 {
+    port: 9292
+};
+
+// Student data service.
+@http:ServiceConfig {
+    basePath: "/records"
+}
+service<http:Service> StudentData bind listener1 {
+
+    int errors = 0;
+    int requestCounts = 0;
+
+    @http:ResourceConfig {
+        methods: ["POST"],
+        path: "/addStudent"
+    }
+    // Add Students resource used to add student records to the system.
+    addStudents(endpoint httpConnection, http:Request request) {
+        // Initialize an empty http response message.
+        requestCounts++;
+        http:Response response;
+        Student stuData;
+
+        // Accepting the Json payload sent from a request.
+        var payloadJson = check request.getJsonPayload();
+
+        //Converting the payload to Student type.
+        stuData = check <Student>payloadJson;
+
+        // Calling the function insertData to update database.
+        json ret = insertData(stuData.name, stuData.age, stuData.mobNo, stuData.address);
+
+        // Send the response back to the client with the returned json value from insertData function.
+        response.setJsonPayload(ret);
+        _ = httpConnection->respond(response);
```

Comment:
Shall we handle the possible errors instead of ignoring them? See https://github.com/ballerina-platform/ballerina-examples/blob/master/examples/hello-world-service/hello_world_service.bal#L14

Reply:
Please fix in other places too.